MainWindow.xaml.cs

```diff
@@ -0,0 +1,936 @@
+using Newtonsoft.Json;
+using RfReader_demo.BAL;
+using RfReader_demo.DAL;
+using RfReader_demo.Helper;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.IO;
+using System.IO.Ports;
+using System.Linq;
+using System.Text;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Documents;
+using System.Windows.Media;
+using System.Xml;
+using static RfReader_demo.BAL._BAL;
+using Sentry;
+using System.Configuration;
+
+namespace RfReader_demo
+{
+    public partial class MainWindow : Window
```

Comment:
There seem to be 3 possible places where configuration data can exist:

- `vt_Common`, which is read directly from the XML file
- `allPortData`, which is stored in memory on the MainWindow class itself
- the form `input` textboxes

There are 2 phases of the application:

- Non-live
- Live

Let's pick some actions, and follow where the configuration data will reside during each of the three phases:

|                                                                                       | Non-Live | Live |
|---------------------------------------------------------------------------------------|----------|------|
| Edit database configuration                                                           |          |      |
| Edit database configuration and click Test Connection without clicking Save           |          |      |
| Edit database configuration and click Save                                            |          |      |
| Add a new device configuration                                                        |          |      |
| Add a new device configuration and click Save                                         |          |      |
| Edit an existing device configuration                                                 |          |      |
| Edit an existing device configuration and click Test Connection without clicking Save |          |      |
| Edit an existing device configuration and click Save                                  |          |      |
| Delete an existing device configuration                                               |          |      |

**For example:** *Edit database configuration (Non-Live)*
**Answer:** Changes will only be in the form `input`?

**Another example:** *Edit database configuration and click Test Connection without clicking Save (Non-Live)*
**Answer:** The edits to the database configuration will be ignored by Test Connection, and previous data will be used?